x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/rare_view/detector_description.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { FC, useContext, useEffect, useState } from 'react';
+import { EuiText, EuiCallOut } from '@elastic/eui';
+
+import { JobCreatorContext } from '../../../job_creator_context';
+import { RareJobCreator } from '../../../../../common/job_creator';
+import { RARE_DETECTOR_TYPE } from './rare_view';
+
+interface Props {
+  detectorType: RARE_DETECTOR_TYPE;
+  isSummary?: boolean;
+}
+
+export const DetectorDescription: FC<Props> = ({ detectorType, isSummary = false }) => {
+  const { jobCreator: jc, jobCreatorUpdated } = useContext(JobCreatorContext);
+  const jobCreator = jc as RareJobCreator;
+  const [description, setDescription] = useState<string | null>(null);
+
+  useEffect(() => {
+    const desc = createDetectorDescription(jobCreator, detectorType, isSummary);
+    setDescription(desc);
+  }, [jobCreatorUpdated]);
+
+  if (description === null) {
+    return null;
+  }
+
+  return isSummary ? (
+    <EuiText>
+      <h5>{description}</h5>
+    </EuiText>
+  ) : (
+    <EuiCallOut
+      title={i18n.translate(
+        'xpack.ml.newJob.wizard.pickFieldsStep.rareField.plainText.calloutTitle',
+        {
+          defaultMessage: 'Detector summary',
+        }
+      )}
+    >
+      {description}
+    </EuiCallOut>
+  );
+};
+
+function createDetectorDescription(
+  jobCreator: RareJobCreator,
+  detectorType: RARE_DETECTOR_TYPE,
+  isSummary: boolean
+) {
+  if (jobCreator.rareField === null) {
+    return null;
+  }
+
+  const rareFieldName = jobCreator.rareField.id;
+  const populationFieldName = jobCreator.populationField?.id;
+  const splitFieldName = jobCreator.splitField?.id;
+
+  const desc = [
```

Comment:
I wonder how easy it would be to build the translation depending on the language given this way of building up the summary? Might it be better to supply each possibility as a single translation, even if the code is not so concise?

Reply:
yes, this might be a better approach for translations.